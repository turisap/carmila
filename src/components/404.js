/**
 * Created by HP on 29-Nov-17.
 */
import React from 'react';
import {Link} from 'react-router-dom';

const PageNotFound = () => (
    <div>
        404 <Link to="/">Go home</Link>
    </div>
);

export default PageNotFound;