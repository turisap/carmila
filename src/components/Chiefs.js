/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';

const Chiefs = () => (
    <div>
        <p>Chiefs page</p>
    </div>
);

export default Chiefs;