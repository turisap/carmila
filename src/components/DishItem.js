/**
 * Created by HP on 10-Dec-17.
 */
import React from 'react';
import {removeItem, removeAllSameItems, incrementItem} from '../actions/orders';
import {connect} from 'react-redux';

export const DishItem = (props) => (
    <div className="dishItem__content">
        <div className="dishItem__cropper">
            <img src={props.path} className="dishItem__img"/>
        </div>
        <h4 className="dishItem__title">{props.title}</h4>
        {props.orderPage ?
            <p><span className="lead">Price:  </span>{props.price * props.amount}</p>
            :
            <p><span className="lead">Price:  </span>{props.price}</p>
        }
        <p className="dishItem__description">{props.description}</p>
        {(props.amount > 0) && <p className="dishItem__amount"><span className="lead">Amount:  </span>{props.amount}</p>}
        {(props.orderPage && props.amount > 0) && (
            <div className="dishItem__buttonHolder">
                <button className="dishitem__removeone btn btn-primary" onClick={() => props.removeItem(props.title)}>-</button>
                <button className="dishitem__removeall btn btn-primary" onClick={() => props.removeAllSameItems(props.title)}>Remove All</button>
                <button className="dishitem__removeone btn btn-primary" onClick={() => props.addItem(props.title)}>+</button>
            </div>
        )}
    </div>
);

const mapDispatchToProps = (dispatch, props) => ({
    removeItem : (title) => dispatch(removeItem(title)),
    removeAllSameItems : (title) => dispatch(removeAllSameItems(title)),
    addItem : (title) => dispatch(incrementItem(title))
});


export default connect(undefined, mapDispatchToProps)(DishItem);