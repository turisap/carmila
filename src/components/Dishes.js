/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';

export default class Dishes extends React.Component{
    render() {
        return (
            <h2>Dishes</h2>
        )
    }
}