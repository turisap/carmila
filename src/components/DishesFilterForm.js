/**
 * Created by HP on 10-Dec-17.
 */
import React from 'react';
import {connect} from 'react-redux';
import {categories} from  '../data/data';
import {setPizzaFilter, setPastaFilter, setDessertFilter, setTextFilter} from '../actions/filters';


export class DishesFilterForm extends React.Component {
    state = {
        activeLink : 'pizza'
    };
    handleCategoryChange = (e) => {
        const filter = e.target.text.toUpperCase();
        if (filter === 'PIZZA') {
            this.props.setPizzaFilter();
            this.setState({activeLink : 'pizza'});
        } else if (filter === 'PASTA') {
            this.setState({activeLink : 'pasta'});
            this.props.setPastaFilter()
        } else if (filter === 'DESSERTS') {
            this.setState({activeLink : 'desserts'});
            this.props.setDessertFilter();
        }
        this.props.setTextFilter('')
    };

    handleFilterTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    };

    render () {
        return (
            <div className="dishes__filtersContainer">
                <input className="form-control dishes___textDishFilter" placeholder="Type what you wish" type="text" onKeyDown={this.handleFilterTextChange}/>

                <div className="dishes__textFiltersContainer">
                    {categories.map((c,i) => {
                        return (
                            <a key={i} className={c.toLowerCase() === this.state.activeLink ? 'dishes--activeCategory dishes__text-filter' : 'dishes__text-filter'} onClick={this.handleCategoryChange}>{c}</a>
                        )
                    })}
                </div>
            </div>
        )
    }
}

const mapDispatchToProps = dispatch => {
    return {
        setPizzaFilter : () => dispatch(setPizzaFilter()),
        setPastaFilter : () => dispatch(setPastaFilter()),
        setDessertFilter : () => dispatch(setDessertFilter()),
        setTextFilter  : (text) => dispatch(setTextFilter(text))
    }
};


export default connect(undefined, mapDispatchToProps)(DishesFilterForm);