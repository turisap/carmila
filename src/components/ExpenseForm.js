/**
 * Created by HP on 01-Dec-17.
 */
import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';


const date = moment().format('MMM Do, YYYY');

export default class ExpenseForm extends React.Component {
    constructor (props) {
        super(props);

        this.state = {
            description : props.expense ? props.expense.description : '',
            note : props.expense ? props.expense.note : '',
            amount : props.expense ? (props.expense.amount / 100).toString() : '',
            createdAt : props.expense ? new Date(moment(props.expense.createdAt).valueOf()) : new Date(),
            error : '',
            calendarFocused : null,
        };
    }

    onDescriptionChange = (e) => {
        this.setState({
            description : e.target.value
        })
    };

    onNotesChange = (e) => {
        this.setState({
            note : e.target.value
        })
    };

    onAmountChange = (e) => {
        const amount = e.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState({
                amount : e.target.value
            })
        }
    };

    onDateChange = (createdAt) => {
        if(createdAt)this.setState({ createdAt : createdAt});
    };

    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.description || !this.state.amount) {
            this.setState({error : 'You haven\' filled amount or description'});
            return;
        }
        this.setState({error : ''});
        this.props.onFormSubmit({
            description : this.state.description,
            amount : parseFloat(this.state.amount) * 100,
            createdAt : moment(this.state.createdAt).valueOf(),
            note : this.state.note
        })
    };

    render() {
        //console.log(this.props.expense.description)
        //console.log(this.state)
        return (
            <div>
                <form onSubmit={this.onSubmit}>
                    {this.state.error && <p>{this.state.error}</p>}
                    <input
                        type="text"
                        placeholder="Description"
                        value={this.state.description}
                        onChange={this.onDescriptionChange}
                    />
                    <input
                        type="text"
                        placeholder="Amount"
                        value={this.state.amount}
                        onChange={this.onAmountChange}
                    />
                    <SingleDatePicker
                        onDateChange={this.onDateChange}
                        focused={this.state.focused} // PropTypes.bool
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                        onFocusChange={({ focused }) => this.setState({ focused })} // PropTypes.func.isRequired
                    />
                    <textarea
                        placeholder="Put noutes about this expense (optional)"
                        onChange={this.onNotesChange}
                        value={this.state.notes}
                    ></textarea>
                    <button>Add Expense</button>
                </form>
            </div>
        )
    }
}