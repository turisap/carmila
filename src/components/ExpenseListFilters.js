/**
 * Created by HP on 01-Dec-17.
 */
import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused : null,
    };
    onDatesChange = ({startDate, endDate}) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };

    onFocusChange = (calendarFocused) => {
        this.setState(() => ({calendarFocused}));
    };

    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    };

    render() {
        return (
            <div>
                <input type="text" value={this.props.filters.text} onChange={this.onTextChange}/>
                <DateRangePicker
                    startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
                    endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
                    onDatesChange={this.onDatesChange} // PropTypes.func.isRequired,
                    focusedInput={this.state.calendarFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                    onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                />
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters,
    }
};

const mapDispatchToProps = (dispatch) => ({
    setTextFilter : (text) => dispatch(setTextFilter(text)),
    setStartDate : (startDate) => dispatch(setStartDate(startDate)),
    setEndDate   : (endDate) => dispatch(setEndDate(endDate))
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);

