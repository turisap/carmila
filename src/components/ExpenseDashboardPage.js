/**
 * Created by HP on 29-Nov-17.
 */
import React from 'react';
import ExpenseList from '../components/ExpenseList';
import ExpenseListFilters from '../components/ExpenseListFilters';

const ExpenseDashBoardPage = () => (
    <div>
        <ExpenseListFilters/>
        <ExpenseList/>
    </div>
);

export default ExpenseDashBoardPage;