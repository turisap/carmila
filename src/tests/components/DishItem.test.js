/**
 * Created by HP on 16-Dec-17.
 */
import React from 'react';
import { shallow } from 'enzyme';
import {DishItem} from '../../components/DishItem';

let wrapper, removeItem, removeAllSameItems, item;

beforeEach(() => {
    item = {
        category: "PIZZA",
        title: 'Al Taglio',
        price: 6.5,
        description : 'test description',
        amount : 1
    };
    removeItem = jest.fn();
    removeAllSameItems = jest.fn();
    wrapper =  shallow(<DishItem
        removeItem={removeItem}
        removeAllSameItems={removeAllSameItems}
        {...item}
        orderPage={true}
    />)
});

test('should render Dish item component', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should remove one item', () => {
   wrapper.find('p.dishitem__removeone').simulate('click');
   expect(removeItem).toHaveBeenCalledWith(item.title);
});

test('should remove all identical items', () => {
    wrapper.find('p.dishitem__removeall').simulate('click');
    expect(removeAllSameItems).toHaveBeenCalledWith(item.title);
});