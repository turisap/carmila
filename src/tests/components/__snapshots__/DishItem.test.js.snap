// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Dish item component 1`] = `
<div
  className="dishItem__content"
>
  <div
    className="dishItem__cropper"
  >
    <img
      className="dishItem__img"
    />
  </div>
  <h4
    className="dishItem__title"
  >
    Al Taglio
  </h4>
  <p>
    <span
      className="lead"
    >
      Price:  
    </span>
    6.5
  </p>
  <p
    className="dishItem__description"
  >
    test description
  </p>
  <p
    className="dishItem__amount"
  >
    <span
      className="lead"
    >
      Amount:  
    </span>
    1
  </p>
  <div
    className="dishItem__buttonHolder"
  >
    <button
      className="dishitem__removeone btn btn-primary"
      onClick={[Function]}
    >
      -
    </button>
    <button
      className="dishitem__removeall btn btn-primary"
      onClick={[Function]}
    >
      Remove All
    </button>
    <button
      className="dishitem__removeone btn btn-primary"
      onClick={[Function]}
    >
      +
    </button>
  </div>
</div>
`;
